sparse-index.c

```diff
@@ -297,7 +378,7 @@ void ensure_full_index(struct index_state *istate)
 	/* Copy back into original index. */
 	memcpy(&istate->name_hash, &full->name_hash, sizeof(full->name_hash));
 	memcpy(&istate->dir_hash, &full->dir_hash, sizeof(full->dir_hash));
-	istate->sparse_index = 0;
+	istate->sparse_index = pl ? PARTIALLY_SPARSE : 0;
```

Comment:
```suggestion
	istate->sparse_index = pl ? PARTIALLY_SPARSE : COMPLETELY_FULL;
```

Same as the other comment - if what's currently there is the right way to write this, you can ignore/resolve this suggestion.